OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -149,16 +149,21 @@ public override bool Tick(Actor self)
 			if (!targetIsHiddenActor && target.Type == TargetType.Actor)
 				lastVisibleTarget = Target.FromTargetPositions(target);
 
-			useLastVisibleTarget = targetIsHiddenActor || !target.IsValidFor(self);
+			// If the target died while hidden, we shouldn't be able to know that, so assume the target is still hidden.
+			if (target.Type == TargetType.Actor)
```

Comment:
The comment here does not make much sense to me. If the target died (even while being hidden) `target.Type` will be `invalid` (see https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Game/Traits/Target.cs#L101-L102).

Reply:
That is exactly the problem that is being addressed here. Which is why we must only update the state when the target is not invalid. If the target dies, we treat it as if it had the visibility state it had when it was last alive.

(Note that `targetIsHiddenActor` is always false when the target is invalid)

Reply:
Yes, and that is why I think we should reword the comment. The way it currently is suggests the if condition is true if the target died while being hidden. What about something like `Only consider updating the visibility state when the target is still valid and visible`?

Reply:
Reworded the comment.